test/parallel_api/ranges/std_ranges_find_end.pass.cpp

```diff
@@ -0,0 +1,34 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "std_ranges_test.h"
+
+std::int32_t
+main()
+{
+#if _ENABLE_STD_RANGES_TESTING
+    using namespace test_std_ranges;
+    namespace dpl_ranges = oneapi::dpl::ranges;
+
+    auto find_end_checker = TEST_PREPARE_CALLABLE(std::ranges::find_end);
+
+    test_range_algo<0, int, data_in_in>{big_sz}(dpl_ranges::find_end, find_end_checker, binary_pred);
+    test_range_algo<1, int, data_in_in>{}(dpl_ranges::find_end, find_end_checker, binary_pred, proj, proj);
```

Comment:
`data_in_in` generates data this way:

```c++
Container cont_in1(exec, max_n, [](auto i) { return i;});
Container cont_in2(exec, max_n, [](auto i) { return i % 5 ? i : 0;});
```

It will always result in false for `find_end` algorithm. The data generation should be adjusted to test "true" cases.

Reply:
Such test cases ha been added.